tasks/wait-for-image-task.yaml

```
@@ -58,6 +46,34 @@ spec:
         exit 1
       fi
 
+      # Output
       echo -n "${infos[0]}" | tee "$(results.IMAGE_DIGEST.path)"
       echo -n "${infos[1]}" | tee "$(results.GIT_REF.path)"
       echo -n "${infos[2]}" | tee "$(results.GIT_REPO.path)"
+
+      IMAGE_DIGEST="${infos[0]}"
+      echo "Waiting for attestation for ${IMAGE}@${IMAGE_DIGEST} to appear..."
+      while true; do
+        if cosign tree "${IMAGE}@${IMAGE_DIGEST}" | grep -qE "quay.io/.*:sha256-.*\.att$"; then
+          break
+        fi
+        echo "Attestation for ${IMAGE}@${IMAGE_DIGEST} not (yet) found, sleeping and retrying..."
+        sleep 1m
+      done
+
+      echo "Attestion for ${IMAGE}@${IMAGE_DIGEST} found, which means PipelineRun finished. Checking if any tasks finished unsuccessfully."
+
+      no_success=$(cosign download attestation "${IMAGE}@${IMAGE_DIGEST}" | \
+        jq -r '.payload' | \
+        base64 -d | \
+        jq '.predicate.buildConfig.tasks | [ .[] | select(.status != "Succeeded") ] | length'
+      )
+
+      if [ "${no_success}" -gt 0 ]; then
+        echo "ERR: At least one task has failed during build of ${IMAGE}@${IMAGE_DIGEST}."
+        echo "ERR: Try building it again and re-run this pipeline."
```

1. [nit] The message above uses `ERROR`, please be consistent (either way).
2. [nit] Also, it could be graceful to echo to stderr (also in both places).
3. Suggestion: let's tell why it's bad.

```suggestion
        >&2 echo "ERROR: At least one task has failed during build of ${IMAGE}@${IMAGE_DIGEST}."
        >&2 echo "ERROR: This image will likely fail Conforma checks. Try building it again and re-run this pipeline."
```